src/providers/mbed_crypto/mod.rs

```diff
@@ -117,8 +117,9 @@ impl Provider {
                                     max_key_id = key_id;
                                 }
                             }
-                            Err(status::Error::DoesNotExist) => to_remove.push(key_triple.clone()),
+                            Err(status::Error::InvalidHandle) => to_remove.push(key_triple.clone()),
                             Err(e) => {
+                                println!("Key ID was {}", key_id);
```

Comment:
I guess you used that to debug. Maybe that should be a `format_error` here.

Reply:
Yeah and nah - will just remove it, I wanted to see if the ID we were getting was actually invalid.